src/app/browser/qgsinbuiltdataitemproviders.cpp

```diff
@@ -878,105 +878,139 @@ QString QgsDatabaseItemGuiProvider::name()
 void QgsDatabaseItemGuiProvider::populateContextMenu( QgsDataItem *item, QMenu *menu, const QList<QgsDataItem *> &selectedItems, QgsDataItemGuiContext context )
 {
   Q_UNUSED( selectedItems )
+
+  // Small helper to get a connection from either a connection name (if it's stored) or a data item path
+  auto getConnectionFromDataItem = [ ]( QgsDataCollectionItem * collectionItem, bool isSchema ) -> QgsAbstractDatabaseProviderConnection *
```

Comment:
Wouldn't it be cleaner/easier to add a virtual "connectionName" property to QgsDataCollectionItem?

Reply:
I don't think so: the problem here is that we can get to that piece of code from two different situations:
1. from an existing stored connection (this was handled correctly)
2. from the file browser (no stored connection available, this was failing)

In case 1 we get the connection from the stored connections but in the second case we need to create a (temporary) connection on the fly based on the file path that we can get from the dataItem->path().

I thought about adding a virtual dataItem->connection() but it would be quite a refactoring and I'm not sure it worth the effort: we only have this problem with file-system based DBs (gpkg and spatialite).

Reply:
Hmm - the current approach seems very fragile to me (and unfortunately it's almost impossible to protect with unit tests). What about a setter/getter for the connection name instead of a virtual method? It's known at the time the database items are created, so we could easily store it then..

Reply:
Ok, I refactored and moved the logic to `QgsDataItem::databaseConnection`, as a bonus we have spatialite working from the filesystem browser too.